src/Psalm/Internal/LanguageServer/Server/Workspace.php

```diff
@@ -0,0 +1,73 @@
+<?php
+declare(strict_types = 1);
+namespace Psalm\Internal\LanguageServer\Server;
+
+use LanguageServerProtocol\FileChangeType;
+use LanguageServerProtocol\FileEvent;
+use Psalm\Codebase;
+use Psalm\Internal\LanguageServer\LanguageServer;
+
+/**
+ * Provides method handlers for all textDocument/* methods
+ */
+class Workspace
+{
+    /**
+     * @var LanguageServer
+     */
+    protected $server;
+
+    /**
+     * @var Codebase
+     */
+    protected $codebase;
+
+    /** @var ?int */
+    protected $onchange_line_limit;
+
+    public function __construct(
+        LanguageServer $server,
+        Codebase $codebase,
+        ?int $onchange_line_limit
+    ) {
+        $this->server = $server;
+        $this->codebase = $codebase;
+        $this->onchange_line_limit = $onchange_line_limit;
+    }
+
+    /**
+     * The watched files notification is sent from the client to the server when the client
+     * detects changes to files and folders watched by the language client (note although
+     * the name suggest that only file events are sent it is about file system events
+     * which include folders as well). It is recommended that servers register for these
+     * file system events using the registration mechanism. In former implementations clients
+     * pushed file events without the server actively asking for it.
```

Comment:
This implies server has to register files it wants to watch, but I don't see it anywhere. So I guess this may break eventually.

Reply:
Yes. I read into that myself. Right now vscode will send the workspace changes without registering which is good

The bigger issue is that the Felix package Psalm relies on seems to be abandonware (not updated since 2018) so there's no ability to register capabilities. 

At this point it's either fork that project and have psalm reference it's own version. Or fork out only the classes Psalm needs

Besides that if the day comes and vscode stops publishing the events unless psalm registers it doesn't break anything internally at that point we will just need to work on the register call which will hopefully be done before that point

As for now vscode works as is and sends the change regardless

Reply:
With the latest release 17 days ago I wouldn't call it abandonware. It would probably suffice to craft a PR and nudge Felix to make a release.

Reply:
I'm mainly speaking of https://github.com/felixfbecker/php-language-server/issues/807

Reply:
Yeah, but we don't use that.

Reply:
Oh yeah. Right. I don’t know why I thought that was used. The ones that are used are pretty up to date and support php 8. I’ll look into the registry calls for a future pr